### PR TITLE
RET-5353

### PR DESCRIPTION
**Pull request by JustMehmet**
### JIRA link

https://tools.hmcts.net/jira/browse/RET-5353

-----

### Change description

Updates for code quality

-----

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/utils/ErrorUtils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  public static throwError(err: Error, errorName: string): void {
+    const error = new Error(err.message);
+    error.name = errorName;
+    if (err.stack) {
+      error.stack = err.stack;
+    }
+    throw error;
+  }
+
+  public static throwManuelError(message: string, name: string): void {
```
Comment (hsjhita) [inline src/main/utils/ErrorUtils.ts:14]:
```suggestion
  public static throwManualError(message: string, name: string): void {
```

File: src/main/utils/ErrorUtils.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    const err = new Error(message);
+    err.name = name;
+    throw err;
+  }
+
+  public static readonly setManuelErrorToRequestSession = (
```
Comment (hsjhita) [inline src/main/utils/ErrorUtils.ts:20]:
```suggestion
  public static readonly setManualErrorToRequestSession = (
```

File: src/test/unit/services/CaseSelectionService.test.ts [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    mockApiClient.updateDraftCase.mockClear();
+    mockApiClient.getUserCase.mockClear();
+  });
+
+  it('Should Return user cases by last modified date', async () => {
+    const response: AxiosResponse<CaseApiDataResponse[]> = {
+      data: [
+        {
+          id: '12234',
+          state: CaseState.AWAITING_SUBMISSION_TO_HMCTS,
```
Comment (hsjhita) [inline src/test/unit/services/CaseSelectionService.test.ts:59]: if I'm being picky, the respondent wouldn't be able to see the case if it's in a draft state, only accepted & closed

File: src/main/utils/ErrorUtils.ts (hunk 1 above)
Comment (JustMehmet) [reply src/main/utils/ErrorUtils.ts:14]: Thanks harpreet :) Corrected

File: src/main/utils/ErrorUtils.ts (hunk 2 above)
Comment (JustMehmet) [reply src/main/utils/ErrorUtils.ts:20]: Thanks Harpreet, corrected :)

**Review by hsjhita — APPROVED** [review]

**Comment by hsjhita** [comment]
@JustMehmet the micromatch CVE issue has been fixed in https://github.com/hmcts/et-syr-frontend/pull/91. Can your yarn-audit files be updated to reflect it? Thanks

**Comment by JustMehmet** [comment]
> @JustMehmet the micromatch CVE issue has been fixed in #91. Can your yarn-audit files be updated to reflect it? Thanks

@hsjhita Thank you so much Harpreet :)

**Review by hsjhita — APPROVED** [review]

**Pull request merged**